vertical-pod-autoscaler/enhancements/7862-cpu-startup-boost/README.md

```diff
@@ -82,10 +82,29 @@ VPA object) without having to ALSO use the traditional VPA functionality.
 1. When a pod targeted by that VPA is created, the kube-apiserver invokes the
 VPA Admission Controller
 
-1. The VPA Admission Controller modifies the pod's containers CPU request and
-limits to align with its `StartupBoost` policy, if specified, during the pod
-creation. The boosted value is based on the VPA recommendation available at the
-time of admission. During the boost period, no resizing will take place.
+
+1.  The VPA Admission Controller modifies the pod's containers CPU request and
+    limits to align with its `StartupBoost` policy, if specified, during the pod
+    creation. The base value for the boost calculation is the VPA recommended CPU
+    request. If the VPA recommendation is not available or is zero, the container's
+    original CPU request from the Pod spec is used as the base.
+
+    The behavior for CPU limits depends on the `ControlledValues` setting in the
+    `ContainerResourcePolicy`:
+
+    *   If `ControlledValues` is `RequestsOnly` (the default), the boosted CPU request
+        will be capped at the container's original CPU limit, if one is set.
```

Comment:
I am not sure this should be the behavior when this policy is used and the original CPU limit is set. With this policy, the admission controller does not set any limits, even when the original limit exists. Using the original CPU limit is not a good idea when there is CPU recommendation available, because the recommendation and the original Pod-specified CPU limit have no meaningful relationship.

My opinion is that when we use `RequestsOnly`- which means VPA does not control limits - the original CPU limit should not be considered when making capping decisions

Reply:
> My opinion is that when we use RequestsOnly- which means VPA does not control limits - the original CPU limit should not be considered when making capping decisions

If we use RequestOnly, the pod has both request and limit set and we only boost the request value, we risk having request > limit. So, I believe that we need to cap the boosted value to the container limits here as well (if set).